OrbitVulkanLayerClientGgp/LayerLogic.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "LayerLogic.h"
+
+#include <math.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitCaptureGgpClient/OrbitCaptureGgpClient.h"
+#include "absl/base/casts.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+#include "absl/synchronization/mutex.h"
+#include "absl/time/time.h"
+
+namespace {
+static constexpr uint16_t kGrpcPort = 44767;
+static constexpr double kFrameTimeThreshold = 1000 / 60.0;  // milliseconds
+static constexpr int64_t kCaptureLength = 10;               // seconds
+}  // namespace
+
+void LayerLogic::StartOrbitCaptureService() {
+  LOG("Starting Orbit capture service");
+  pid_t pid = fork();
+  if (pid < 0) {
+    ERROR("Fork failed; not able to start the capture service");
+  } else if (pid == 0) {
+    std::string game_pid = absl::StrFormat("%d", getppid());
+    // TODO(crisguerrero): Read the arguments from a config file.
+    char* argv[] = {strdup("/mnt/developer/OrbitCaptureGgpService"),
```

Comment:
Using strdup is a memory leak. If allocates a copy of the string that is never freed.

Reply:
Oh I was not aware of that; I have removed it. Thanks!